flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java

```diff
@@ -384,6 +358,22 @@ public static boolean stateBackendFromApplicationOrConfigOrDefaultUseManagedMemo
         return false;
     }
 
+    /**
+     * Try to unwrap a DelegatingStateBackend and get the inner delegated StateBackend. If the
+     * provided StateBackend is not a delegated StateBackend, just return itself.
+     *
+     * @param backend the provided StateBackend that may delegate another StateBackend.
+     * @return the root StateBackend W/O delegation.
+     */
+    @VisibleForTesting
+    public static StateBackend unwrapFromDelegatingStateBackend(StateBackend backend) {
+        if (backend != null && backend instanceof DelegatingStateBackend) {
```

Comment:
nit: Is this method used **only** for tests? If so, maybe move it to `test` source directory?
nit: I think `null` check is unnecessary here.

Reply:
Sure, moved to ```ChangelogStateBackendLoadingTest```.